src/forder.c

```diff
@@ -1419,3 +1463,252 @@ SEXP binary(SEXP x)
   return ans;
 }
 
+// all(x==1L)
+static bool all1(SEXP x) {
+  if (!isInteger(x))
+    error("internal error: all1 got non-integer"); // # nocov
+  int *xp = INTEGER(x);
+  for (int i=0; i<LENGTH(x); ++i) {
+    if (xp[i] != 1)
+      return false;
+  }
```

Comment:
nit: any reason to put this on 4 lines? seems like it's a simple enough function that we should just do

```suggestion
  for (int i=0; i<LENGTH(x); ++i) if (xp[i] != 1) return false;
```

Reply:
Ensuring codecov will be well covering each piece of code

Reply:
i would say overkill... function is helper & pretty trivial, unless you want to add NA handling then it will be more natural to use up so much space. codecov saying "this helper was indeed used" would be good enough for me.